dkg-gadget/src/worker.rs

```diff
@@ -1383,16 +1405,19 @@ where
 						break;
 					}
 				},
-
-				dkg_msg = dkg.next().fuse() => {
-					if let Some(dkg_msg) = dkg_msg {
-						log::debug!(target: "dkg_gadget::worker", "Going to handle dkg message for round {}", dkg_msg.msg.round_id);
-						self.process_incoming_dkg_message(dkg_msg);
-					} else {
-						log::error!("DKG stream closed");
-						break;
-					}
-				},
+			}
+			// Finally, Dequeue a message from the dkg message queue.
+			let next_dkg_message = self.gossip_engine.dequeue_message();
```

Comment:
Maybe consider re-architecting this a bit? If we have the rounds objects for this message or even for the queue entirely (if we make queues on a per round basis), then we should `drain` the queue and process all the messages.

Reply:
I understand that this is being looped over, we can maybe make it faster by just draining it in one go.